src/camera_utils.py

```
@@ -65,6 +65,97 @@ def compute_image_footprint_on_surface(camera: Camera, distance_from_surface: fl
+    footprint_x = footprint.pt_2.x - footprint.pt_1.x
+    footprint_y = footprint.pt_2.y - footprint.pt_1.y
+
+    return np.array([footprint_x, footprint_y])
+
+
+def compute_image_footprint_on_surface_coord(camera: Camera,
+                                             distance_from_surface: float,
+                                             angle_x_deg: float = 0,
+                                             angle_y_deg: float = 0) -> FootprintCoord:
+    """Compute the footprint's coordinate relative to the camera's position,
+    where the footprint is the image captured by the camera 
+    at a given distance from the surface.
+
+    It is assumed that only one of angle_x_deg and angle_y_deg may be non-zero.
+    If both are non-zero, then only angle_x_deg will be used, and angle_y_deg will be 
+    ignored as if it is set to zero.
+
+    Args:
+        camera (Camera): the camera model.
+        distance_from_surface (float): distance from the surface (in m).
+        angle_x_deg (float): camera's gimbal angle toward the x direction (in degrees)
+        angle_y_deg (float): camera's gimbal angle toward the y direction (in degrees)
+
+    Returns:
+        FootprintCoord: the coordinates of the footprint relative to the camera's position.
+        Note: Since the camera is aiming downward toward the ground, the z coordinate of 
+        a point on the ground will be equal to -distance_from_surface, which is a negative value.
+
+        Note: A simplifying assumption is made that the footprint will remain a rectangular region,
+        But with non-zero gimbal angle, there will be a distortion so the footprint may not actually 
+        be rectangular in reality.
+    """
+
+    # convert to radians
+    angle_x = math.radians(angle_x_deg) # convert deg to rad
+    angle_y = math.radians(angle_y_deg) # convert deg to rad
+
+    # Check to make sure that at most one angle is non-zero
+    # If both are non-zero, then set angle_y = 0.
+    try:
```

Curious why raise an exception and immediately catch it. If you do not want to bubble up the error, just use an if-else. Otherwise, bubble up the error to the caller.

---

Modified as suggested, to simply let the error bubble up. The new version will simply raise an exception when both camera angles are non-zero, rather than setting angle_y to zero.